app/services/users/omniauth.rb

```diff
@@ -6,31 +6,37 @@ def initialize(auth)
   end
 
   def get
-    user
+    check_auth
+    @user
   end
 
   private
 
   delegate :provider, :uid, :info, to: :auth
   delegate :email, to: :info
 
-  def user
-    @_user ||= find_user || create_user
+  def find_authorization(user)
+    user.authorizations.where(provider: provider, uid: uid).first
   end
 
-  def find_user
-    User.where(provider: provider, uid: uid, email: email).first
+  def create_authorization(user)
+    user.authorizations.create(provider: provider, uid: uid)
+  end
+
+  def initiate_user
+    @user = User.where(email: email).first || create_user
+  end
+
+  def check_auth
+    initiate_user
+    create_authorization(@user) if find_authorization(@user).blank?
   end
 
   def create_user
-    user = User.new(
-      provider: provider,
-      uid: uid,
+    User.create(
       email: email,
-      password: Devise.friendly_token[0, 20]
+      password: Devise.friendly_token[0, 20],
+      confirmed_at: Date.current
     )
-    user.skip_confirmation!
-    user.save
-    user
   end
 end
```

Comment:
@sinpai so, after all suggestions for this service I see:
```ruby
class Users::Omniauth < Struct.new(:auth)
  def get
    find_or_create_authorization
    user
  end

  private

  def find_or_create_authorization
    user.authorizations
      .create_with(uid: uid)
      .find_or_create_by(provider: provider)
  end
  
  def user
    @_user ||= User.find_or_create_by(email: email) do |user|
      user.password = generate_password
      user.confirmed_at = Date.current
    end
  end

  def generate_password
    Devise.friendly_token[0, 20]
  end

  delegate :provider, :uid, :info, to: :auth
  delegate :email, to: :info
end
```

Reply:
@sinpai what do you think about updating tests for this service?

Reply:
👍 Looks very cool! Thanks!